src/scales/scale.time.js

```diff
@@ -358,23 +358,23 @@ function generate(scale) {
  * They add extra margins on the both sides by scaling down the original scale.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, timestamps, min, max, options) {
 	let start = 0;
 	let end = 0;
 	let first, last;
 
-	if (options.offset && ticks.length) {
-		first = interpolate(table, 'time', ticks[0], 'pos');
-		if (ticks.length === 1) {
+	if (options.offset && timestamps.length) {
```

Comment:
I still don't like it, because it depends on the data intervals. If first or last data interval happens to be a big leap, then there is going to be a big offset.
`offset = 1  / timestamps.length` would work consistently, although IMO the offset option is about `ticks` and not `data`, so I'd prefer `1 / ticks.length`.

Reply:
I'm really confused why you think it should depend on ticks and not data. I added some screenshots and additional text to the PR description that hopefully demonstrate why I think it needs to depend on data. Let me know if it doesn't make sense.

As to changing the sizing, I'd rather not change that in this PR because you can have bars that are different widths. I think it was done this way originally to account for that case. If the last bar is much bigger than the first one, then it makes sense that the offset on the right edge would be much bigger. I think the notion of having differently sized bars is a bit odd and haven't seen that used much if at all, but I think we'd have to remove that functionality and force all bars to be the same size if we wanted the offsets to also be the same size. That's really beyond the scope of this PR where I'm just trying to fix the ticks vs data issue

Reply:
Actually thinking about (maybe a more common) use case, line chart and want to use the width for the chart rather than for the right padding the tick needs. So offset: true.
For bar charts, its obviously data count that matters. But that should be the tick count too, shouldn't it?
For line charts, its probably the last ticks texts width / 2 that would work as expected.

But that goes a bit too far, I think this one is a good step to the right direction :)

Reply:
> Actually thinking about a different use case, line chart with lots of points (say 50k) and offset: true.

Yeah, that was sort of where I originally found this bug (though I think it was only 600 points). See the image in https://github.com/chartjs/Chart.js/issues/7071. This PR makes that case work a lot better

> For bar charts, its obviously data count that matters. But that should be the tick count too, shouldn't it?

There can be more bars/data than ticks quite often, which is the issue. See the images in the PR description where this is the case

Reply:
I think @kurkle's original concern here might be what I found in #7415